storage/sqlStorages/mysql/datasource.go

```diff
@@ -0,0 +1,79 @@
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/cube2222/octosql"
+	"github.com/cube2222/octosql/storage/sqlStorages"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+
+	"github.com/cube2222/octosql/config"
+	"github.com/cube2222/octosql/physical"
+)
+
+type MySQLTemplate struct{}
+
+func (t MySQLTemplate) GetAvailableFilters() map[physical.FieldType]map[physical.Relation]struct{} {
+	return map[physical.FieldType]map[physical.Relation]struct{}{
+		physical.Primary: {
+			physical.Equal:        {},
+			physical.NotEqual:     {},
+			physical.MoreThan:     {},
+			physical.LessThan:     {},
+			physical.GreaterEqual: {},
+			physical.LessEqual:    {},
+			physical.Like:         {},
+		},
+		physical.Secondary: {
+			physical.Equal:        {},
+			physical.NotEqual:     {},
+			physical.MoreThan:     {},
+			physical.LessThan:     {},
+			physical.Like:         {},
+			physical.GreaterEqual: {},
+			physical.LessEqual:    {},
+		},
+	}
+}
+
+func (t MySQLTemplate) GetIPAddress(dbConfig map[string]interface{}) (string, int, error) {
+	return config.GetIPAddress(dbConfig, "address", config.WithDefault([]interface{}{"localhost", 3306}))
+}
+
+func (t MySQLTemplate) GetConnection(user, password, host, dbName string, port int) (*sql.DB, error) {
```

Comment:
Change this to GetDSN and return the connection string. Then use sql.Open in the generic part on that DSN.

Reply:
DSN + Driver